topics/wallet-interaction-standard.md

```diff
@@ -0,0 +1,265 @@
+
+#### Types
+
+- `text`: A plain `string` value.
+- `blob`: A `string` value describing binary data encoded in base64.
+
+#### Request
+
+`version` (`text`): The version of the standard used.
+
+`appMetadata`: Information about the relying party.
+- `name` (`text`): The user-friendly name of the relying party. 
+- `icon` (`text`, optional): An optional URL pointing to an icon resource representing the relying party.
+
+`networks`: A list of networks on which the relying party plans to operate.
+- `chainId`(`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes the relying party requires. Possible values: 
+- `"canister_call"`
+
+`challenge` (`blob`): A challenge used for the wallet to sign in order to prove its access to the identity. The challenge should be an array of 32 cryptographically random bytes generated by the sender of the request.
+
+#### Response
+
+`version` (`text`): The version of the standard used.
+
+`appMetadata`: Information about the wallet.
+- `name` (`text`): The user-friendly name of the wallet.
+- `icon` (`text`, optional): An optional URL poiting to an icon resource representing the wallet.
+
+`networks`: A list of networks on which the user has agreed the relying party can operate. This should be a subset of the `networks` from the original request.
+- `chainId` (`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes that the user has agreed the relying party can be granted. This should be a subset of the `scopes` field from the original request. Possible values:
+- `"canister_call"`
+
+`identity`: The identity the user has selected.
+- `algorithm` (`text`): The cryptographic algorithm used to generate the key pair associated with the identity. The supported values correspond to signature schemes [supported by the IC](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures) and include:
+    - `"ed25519"`
+    - `"secp256r1"`
+    - `"secp256k1"`
+- `publicKey` (`blob`): The DER-encoded public key associated with the identity. The public key can be used to [derive a self-authenticating principal](https://internetcomputer.org/docs/current/references/ic-interface-spec/#principal).
+- `ledger` (optional): Ledger data.
+    - `subaccount` (`blob`, optional)
+
+`challenge` (`blob`): The challenge the wallet had to sign.
+
+`signature` (`blob`): The signature produced by signing the challenge with the private key associated with the `identity`.
+
+#### Use-Case
+
+1. The relying party sends a `permission` request to the wallet.
```

Comment:
I would add a note that the `challenge` has to be generated from a secure random source.